hipify_torch/cuda_to_hip_mappings.py

```diff
@@ -8302,6 +8302,12 @@
     ]
 )
 
+CUPY_SPECIAL_MAPPINGS = collections.OrderedDict(
+    [
+        ("CUPY_USE_GEN_HIP_CODE", ("CUPY_DONT_USE_GEN_HIP_CODE", API_CUPY)),
```

Comment:
This is the only mapping I'm not too keen on adding to hipify-torch repo, because this is really due to a quirk in how cupy source code is structured, so it doesn't really belong in the hipify-torch repo. @lcskrishna, since you're also working on hipify-torch, it might be a good idea to allow users to extend the hipify-torch mappings by providing their own mappings in a json format (have a default name for that json mappings file), and then invoking hipify_python.py with the mappings file provided. And if hipify_python.py will look for (and use) the default user-provided mapping file, that'll allow even users of CUDAExtension class to provide their own mappings when hipifying their extensions (since they wouldn't need to modify this call: https://github.com/pytorch/pytorch/blob/main/torch/utils/cpp_extension.py#L1077

Reply:
@jithunnair-amd This seems like a good idea to extend the cuda_to_hip_mapping for users like in this case. Let me work on this and update the following PR to reflect that.